src/main.cpp

```diff
@@ -33,7 +36,7 @@ PYBIND11_MODULE(cmake_example, m) {
     )pbdoc");
 
 #ifdef VERSION_INFO
-    m.attr("__version__") = VERSION_INFO;
+    m.attr("__version__") = MACRO_STRINGIFY(VERSION_INFO);
```

Comment:
Is there a reason to not quote this anymore in `setup.py`?

Reply:
Because it can trip up Windows (see the change in python_example and the work being done in PyTorch). And before we were trying to pass a raw flag though by setting a compiler flag directly, which is bad and potentially compiler specific.

Reply:
OK, yes, sounds like a very decent reason :-)

Reply:
Thanks!

Reply:
Macros are bad and ugly. I don't understand why you would want to set the version in a cpp file, when you could set it somewhere in Python.

Reply:
Not every project has a pure Python wrapper around it. To me, that's even the strength of pybind11, that you can do everything in one place (i.e., C++)

Reply:
This is showing how to pass something from Python to C++ via CMake. It's an exact copy of the same thing in python_example. To your point, though, it probably needs a little note saying it is an example, and not necessary a good idea or how you should make your version available.

Reply:
Added a small note.